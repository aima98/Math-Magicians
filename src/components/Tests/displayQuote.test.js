import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DisplayQuote from '../displayQuote';

describe('test page rendering', () => {
  it('test page rendering Quote Title', () => {
    render(
      <DisplayQuote />,
    );

    const quoteTitle = screen.queryByText('Quote');
    expect(quoteTitle).toBeInTheDocument();
  });

  it('test page rendering Quote existence', () => {
    render(
      <DisplayQuote />,
    );

    const quoteTitle = screen.queryByText('Quote');
    expect(quoteTitle.parentElement.querySelector('p')).toBeInTheDocument();
  });
});

describe('snapshot testing', () => {
  test('should persist', () => {
    const quoteTree = renderer.create(
      <DisplayQuote />,
    );
    expect(quoteTree).toMatchSnapshot();
  });
});
