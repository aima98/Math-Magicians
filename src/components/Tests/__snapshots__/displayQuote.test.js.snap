// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should persist 1`] = `
<section
  className="quote_display"
>
  <h2>
    Quote
  </h2>
  <p>
    Loading quote...
  </p>
</section>
`;
